src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -166,8 +166,11 @@ public class HdfsSinkConnectorConfig extends StorageSinkConnectorConfig {
   private static final String KERBEROS_TICKET_RENEW_PERIOD_MS_DISPLAY = "Kerberos Ticket Renew "
       + "Period (ms)";
 
-  public static final String DISABLE_WAL_CONFIG = "wal.disable";
-  private static final boolean DISABLE_WAL_DEFAULT = false;
+  public static final String WAL_TYPE = "wal.type";
+  public static final String WAL_NOOP = "NOOP";
```

Comment:
Could WAL_TYPE be an enum with encoded knowledge about how to construct each.

Add a `WalType getWalType()` method to HDFSConnectorConfig.

```
enum WalType {
  WalType(BuilderFunction<Storage, logsDir,Tp,WAL> factoryMethod) {
  }
  NOOP((s,l,t) -> new NoopWal()),
  QFS_WAL((s,l,t) -> s.wal(l, t, true)),
  HDFS_WAL(s,l,t) -> s.wal(l,t,false));
}
```

then the code TopicPartitionWriter code becomes.
```
 wal = config.getWalType().create(storage, logs, tp)
```